runtime/compiler/control/JitDump.cpp

```diff
@@ -338,81 +304,46 @@ runJitdump(char *label, J9RASdumpContext *context, J9RASdumpAgent *agent)
 
-   // if jitConfig is gone, then we can't do anything either
-   if (jitConfig == 0)
+   J9JITConfig *jitConfig = crashedThread->javaVM->jitConfig;
+   if (NULL == jitConfig)
       {
-      jitDumpFailedBecause(crashedThread, "jitConfig is null");
-      return OMR_ERROR_NONE;
+      j9nls_printf(PORTLIB, J9NLS_ERROR | J9NLS_STDERR, J9NLS_DMP_ERROR_IN_DUMP_STR, "JIT", "Could not locate J9JITConfig");
+      return OMR_ERROR_INTERNAL;
       }
-   // get global compinfo
+      
    TR::CompilationInfo *compInfo = TR::CompilationInfo::get(jitConfig);
-   if (!compInfo)
+   if (NULL == compInfo)
       {
-      jitDumpFailedBecause(crashedThread, "compInfo is null");
-      return OMR_ERROR_NONE;
+      j9nls_printf(PORTLIB, J9NLS_ERROR | J9NLS_STDERR, J9NLS_DMP_ERROR_IN_DUMP_STR, "JIT", "Could not locate TR::CompilationInfo");
+      return OMR_ERROR_INTERNAL;
       }
 
-   // Must be able to allocate a frontend for this thread
    TR_J9VMBase *frontendOfThread = TR_J9VMBase::get(jitConfig, crashedThread);
-   if (!frontendOfThread)
+   if (NULL == frontendOfThread)
       {
-      jitDumpFailedBecause(crashedThread, "thread's frontend is missing");
-      return OMR_ERROR_NONE;
+      j9nls_printf(PORTLIB, J9NLS_ERROR | J9NLS_STDERR, J9NLS_DMP_ERROR_IN_DUMP_STR, "JIT", "Could not locate TR_J9VMBase");
+      return OMR_ERROR_INTERNAL;
       }
 
-   // get global configuration options
    TR::Options *options = TR::Options::getCmdLineOptions();
-   if (!options)
+   if (NULL == options)
       {
-      jitDumpFailedBecause(crashedThread, "No cmdLineOptions available");
-      return OMR_ERROR_NONE;
+      j9nls_printf(PORTLIB, J9NLS_ERROR | J9NLS_STDERR, J9NLS_DMP_ERROR_IN_DUMP_STR, "JIT", "Could not locate TR::Options");
+      return OMR_ERROR_INTERNAL;
       }
 
-
-   // open log file, using the postfixed timestamp if specified
-   TR::FILE *logFile;
-   char tmp[1025];
-   label = frontendOfThread->getFormattedName(tmp, 1025, label, NULL, false);
-   logFile = trfopen(label, "ab", false);
-
    trfprintf(logFile,
       "<?xml version=\"1.0\" standalone=\"no\"?>\n"
       "<jitDump>\n"
       );
 
 #if defined(J9VM_OPT_JITSERVER)
-   if (context && context->javaVM && context->javaVM->jitConfig)
+   if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
       {
-      J9JITConfig *jitConfig = context->javaVM->jitConfig;
-      TR::CompilationInfo *compInfo = TR::CompilationInfo::get(context->javaVM->jitConfig);
-      if (compInfo &&
-          compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
-         {
-         jitDumpFailedBecause(crashedThread, "Skipped jitdump at the JITServer");
-         trfprintf(logFile, "Skipped jitdump at the JITServer\n");
-         trfclose(logFile);
-         return OMR_ERROR_NONE;
-         }
+      return OMR_ERROR_NONE;
```

Comment:
Not 100% familiar what happens when there is a crash on JITServer server side. I'm trying to avoid printing non-IL relating things to the jitdump (some future work in mind). Do you think an NLS print similar to others will suffice here?

Reply:
@dmitry-ten do you have any thoughts on this?

Reply:
I do not understand jitdumps very well, but from the code, I don't see a reason why it can't run on the server just like in regular JIT. You should be able to call `traceILOfCrashedThread` just fine.

Reply:
Hm, maybe adding JITServer support is better done in another PR? Seems like there'd be a whole bunch of subtleties involved. I was more curious as to whether you guys found the message (that's removed in this diff) useful.

Reply:
Can't say that I ever noticed that message before, so I think it's ok to remove it.